include/deal.II/lac/precondition.h

```diff
@@ -986,7 +1047,13 @@ class PreconditionChebyshev : public Subscriptor
   mutable VectorType update2;
 
   /**
-   * Stores the additional data provided to the initialize function.
+   * Internal vector used for the <tt>vmult</tt> operation.
+   */
+  mutable VectorType update3;
```

Comment:
I will add a mutex inside the whole `vmult` function because the same applies to `update1` and `update2`. This class is simply not designed to be used in parallel - one needs to provide several such objects in case one wants to do that.